### DISCUSSION TITLE
Utility function to return page type as enum

### DISCUSSION THREAD

**Discussion by aclist** [aclist/kbin-megamod#279] [Active discussions]
@Pamasich 

Here is what I observed so far. Unlike the mod to redirect sort method to specific page routes, since the majority of mods just need to apply some changes based on the structure of the page, it seems to me that pages fall into certain types of "groups," so I don't think we need to return every atomic page type (e.g. a user page that is a collection page -- this has the same appearance as any other collection page, so whether or not it is a child of a user page seems immaterial here). 

---
|Path|Page type|Notes
|---|---|---|
|/microblog|microblog|the string `microblog` can be located anywhere in the path|
|/people|Users index||
|/magazines|the global magazines index||
|/magazines/collections|Collections index|page style is distinct from magazine index|
|/m/any|an atomic magazine index||
|/c/any|collection index|functionally the same as magazine styling. can also appear on user pages|
|/m/any/t/|thread|if `/t/` is anywhere in the path, it is a thread|
|/newMagazine|create new magazine|style is distinct from `/new/<any>`|
|/new/any|create any other type of content||

Settings
---
|Path|Page type|Notes
|---|---|---|
|/settings/any|settings page||

Notifications
---
|Path|Page type|Notes
|---|---|---|
|/settings/notifications|general notifications||
|/profile/messages/any|inbox||

Everything
---
|Path|Page type|Notes
|---|---|---|
|/all|All threads||
|/|Top page||
|/*|All Content||
|/sub|top page?||
|/sub/microblog|microblogs for everything|same as microblog type above|
|/*/sub|all content for everything||
|/*/m/any|all content for a given magazine||

User pages
---
|Path|Page type|Notes
|---|---|---|
|/u/any/overview|user summary||
|/u/any|user summary|(same as overview page)|
|/u/any/subscriptions|self subs|
|/u/any/threads|user posted threads||
|/u/any/comments|user comments||
|/u/any/posts|user posts||
|/u/any/boosts|user upvotes||

All of these pages functionally share the same styling. In my opinion, these should uniformly return `PAGE.User`.

The same applies to variants like the below:

- /u/any/c/any
- /u/any/c/any/microblog
- /u/any/c/any/*

Additional explanation needed
---

|Path|Page type|Notes
|---|---|---|
|/d/any|?||
|/tag/any|?||

Wasn't sure how to access/find the pages above.

Possibly missing from this list is the Activity page, which is self-contained within a thread but differs from the thread itself.

MBIN (todo: needs more investigation)
---
/home - the top page

**Comment by Pamasich** [comment]
> Path | Page type | Notes
> -- | -- | --
> /d/any | ? |  
> /tag/any | ? |  
>
> Wasn't sure how to access/find the pages above.

/d/ is a domain page, like [/d/google.com](https://kbin.social/d/google.com). It shows all the threads linking to google.com. A fediverse domain like [/d/lemmy.world](https://kbin.social/d/lemmy.world) additionally displays all the non-link threads from that domain.

They have a Threads and a Comments tab. Not sure what the latter is for, it seems to only display one comment chain per thread — and it's still about whether the thread links to the domain, not the comments.

/tag/ shows all the posts using a specific hashtag. Like [#fediverse](https://kbin.social/tag/fediverse) for example.
It has a Threads and a Microblog tab, but by default it actually uses neither. The default view, which isn't manually accessible over the UI, seems to be an alternate microblog view with different functionality than the actual Microblog tab.

Default /tag/ view:
![image](https://github.com/aclist/kbin-kes/assets/18553452/a59e9b4b-3e1f-4d1b-97cd-13da6e878dc0)

/tag/'s microblog view:
![image](https://github.com/aclist/kbin-kes/assets/18553452/e270b094-4283-48b9-adec-2298d06d64df)

The All Content button there just links to the main kbin.social/* page, it's not related to the /tag/ route.

**Comment by aclist** [comment]
This feature is now available via the helper function `getPageType()` and is documented at https://github.com/aclist/kbin-kes/pull/404
